src/style-spec/validate_mapbox_api_supported.js

```diff
@@ -0,0 +1,172 @@
+// @flow
+
+import validateStyle from './validate_style.min';
+import {v8} from './style-spec';
+import readStyle from './read_style';
+import ValidationError from './error/validation_error';
+import getType from './util/get_type';
+
+const SUPPORTED_SPEC_VERSION = 8;
+const MAX_SOURCES_IN_STYLE = 15;
+
+function isValid(value: ?string, regex: RegExp): boolean {
+    if (!value || getType(value) !== 'string') return true;
+    return !!value.match(regex);
+}
+
+function getSourceCount(source: Object) {
+    if (source.url) {
+        return source.url.replace('mapbox://', '').split(',').length;
```

Comment:
Does the first instance of `'mapbox://'` need to be removed before `split`ting? I think the `.replace` call could be removed probably.